frontend/javascripts/oxalis/model/reducers/volumetracing_reducer.ts

```diff
@@ -53,24 +53,30 @@ import type {
   Segment,
   SegmentGroup,
   SegmentMap,
+  TreeGroup,
   VolumeTracing,
 } from "oxalis/store";
 import {
   findParentIdForGroupId,
   getGroupNodeKey,
 } from "oxalis/view/right-border-tabs/trees_tab/tree_hierarchy_view_helpers";
 import type { AdditionalCoordinate, ServerVolumeTracing } from "types/api_types";
-import { mapGroups } from "../accessors/skeletontracing_accessor";
+import { mapGroups, mapGroupsToGenerator } from "../accessors/skeletontracing_accessor";
 import { sanitizeMetadata } from "./skeletontracing_reducer";
+import { forEachGroups } from "./skeletontracing_reducer_helpers";
 
 type SegmentUpdateInfo =
   | {
       readonly type: "UPDATE_VOLUME_TRACING";
       readonly volumeTracing: VolumeTracing;
+      readonly segments: SegmentMap;
+      readonly segmentGroups: TreeGroup[];
     }
   | {
       readonly type: "UPDATE_LOCAL_SEGMENTATION_DATA";
       readonly layerName: string;
+      readonly segments: SegmentMap;
+      readonly segmentGroups: [];
```

Comment:
these properties are added to the `SegmentUpdateInfo` helper type, because they are useful.